src/main/java/org/apache/maven/shared/utils/cli/CommandLineUtils.java

```diff
@@ -288,31 +279,13 @@ public Integer call()
                     errorPumper.setName( "StreamPumper-systemErr" );
                     errorPumper.start();
 
-                    int returnValue;
-                    if ( timeoutInSeconds <= 0 )
+                    if ( timeoutInSeconds > 0 && !p.waitFor( timeoutInSeconds, TimeUnit.SECONDS ) )
```

Comment:
Where is the kill for this process? It may continue to run as a zombie forever.

Reply:
There is defined `ShutDownHook` lines 237 - 253, and it is also called in line 362

There are also `pumper` threads which consume all outputs of process.